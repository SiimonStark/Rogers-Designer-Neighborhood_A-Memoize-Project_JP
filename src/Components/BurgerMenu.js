import React, { Component } from 'react';

class BurgerMenu extends Component {
  constructor(props) {
    super(props)
    this.state = {
    }
  }
  reloadPage = () => {
    window.location.reload(true);
  }
  displayAlert = () => {
    alert('Feature Coming Soon');
  }

  render() {
    
    return (
      <aside className="BurgerMenu" onMouseLeave={this.props.mouseLeave}>
        <ul>
          <li onClick={this.reloadPage}>Reset Game</li>
          <li onClick={this.displayAlert}>Skip Prompt</li>
          <li onClick={this.displayAlert}>Save Game</li>
          <li>
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/grid"
              target="blank"
            >
              Docs: Grid
            </a>
          </li>
          <li>
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/flex"
              target="blank"
            >
              Docs: Flex
            </a>
          </li>
        </ul>
      </aside>
    );
  }
}

export default BurgerMenu;