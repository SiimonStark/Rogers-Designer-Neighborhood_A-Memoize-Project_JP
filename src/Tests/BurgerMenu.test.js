import React from 'react';
import BurgerMenu from '../Components/BurgerMenu';
import { shallow } from 'enzyme';

let mouseLeaveMock = jest.fn();

describe('BurgerMenu', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <BurgerMenu mouseLeave={mouseLeaveMock} />);
  });

  it('should match screenshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should invoke reloadPage() onClick of Li', () => {
    wrapper.instance().reloadPage();
    // Can't test window object
  });
  
  it('should invoke displayAlert() onClick of Li', () => {
    wrapper.instance().displayAlert();
    // Can't test window object
  });
  
  it('should have clickable links', () => {
    wrapper.find('.docsGrid').simulate('click');
  });
  
  it('should have clickable links', () => {
    wrapper.find('.docsFlex').simulate('click');
  });
});