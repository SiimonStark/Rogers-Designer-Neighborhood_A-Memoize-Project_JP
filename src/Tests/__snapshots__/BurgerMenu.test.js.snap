// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu should match screenshot 1`] = `
<aside
  className="BurgerMenu"
  onMouseLeave={[MockFunction]}
>
  <ul>
    <li
      onClick={[Function]}
    >
      Reset Game
    </li>
    <li
      onClick={[Function]}
    >
      Skip Prompt
    </li>
    <li
      onClick={[Function]}
    >
      Save Game
    </li>
    <li>
      <a
        className="docsGrid"
        href="https://developer.mozilla.org/en-US/docs/Web/CSS/grid"
        target="blank"
      >
        Docs: Grid
      </a>
    </li>
    <li>
      <a
        className="docsFlex"
        href="https://developer.mozilla.org/en-US/docs/Web/CSS/flex"
        target="blank"
      >
        Docs: Flex
      </a>
    </li>
  </ul>
</aside>
`;
